query-translator/src/translate.rs

```diff
@@ -68,6 +73,40 @@ impl ToConstraint for ColumnConstraint {
                 Constraint::equal(qa.to_column(), ColumnOrExpression::Value(tv)),
             EqualsColumn(left, right) =>
                 Constraint::equal(left.to_column(), right.to_column()),
+            EqualsPrimitiveLong(table, value) => {
+                let value_column = QualifiedAlias(table.clone(), DatomsColumn::Value).to_column();
+                let tag_column = QualifiedAlias(table, DatomsColumn::ValueTypeTag).to_column();
+                Constraint::And {
+                    constraints: vec![
+                        Constraint::equal(value_column, ColumnOrExpression::Value(TypedValue::Long(value))),
+                        Constraint::In {
+                            left: tag_column,
+                            list:
+                                if value < 0 {
```

Comment:
Consider finding a way to encode these symbolically, referencing the `ValueType` type in some way.  (Rust does not make this as pleasant as Swift.)